src/AggregateFunctions/ReservoirSamplerDeterministic.h

```diff
@@ -220,10 +217,19 @@ class ReservoirSamplerDeterministic
 
     void insertImpl(const T & v, const UInt32 hash)
     {
+        if (!good(hash))
+            return;
+
         /// Make a room for plus one element.
         while (samples.size() >= max_sample_size)
+        {
             setSkipDegree(skip_degree + 1);
 
+            /// Still good?
```

Comment:
This is the fix.

In previous version, the value is not being filtered after coarsening, so one excessive value can appear in the final reservoir.